content/en/blogs/what-are-kubernetes-operators.md

```diff
@@ -0,0 +1,139 @@
+
+
+## CRDs in Kubernetes
+
+As mentioned above, CRDs define the behaviors of a custom resource, such as its desired state as well as the topology of the object. Let’s take a look at an example taken from the [CNCF whitepaper](https://github.com/cncf/tag-app-delivery/blob/eece8f7307f2970f46f100f51932db106db46968/operator-wg/whitepaper/Operator-WhitePaper_v1-0.md):
+
+
+```
+apiVersion: example-app.appdelivery.cncf.io/v1alpha1
+kind: ExampleApp
+metadata:
+  name: appdelivery-example-app
+spec:
+  appVersion: 0.0.1
+  features:
+    exampleFeature1: true
+    exampleFeature2: false
+  backup:
+    enabled: true
+    storageType: "s3"
+    host: "my-backup.example.com"
+    bucketName: "example-backup"
+status:
+  currentVersion: 0.0.1
+  url: https://myloadbalancer/exampleapp/
+  authSecretName: appdelivery-example-app-auth
+  backup:
+    lastBackupTime: 12:00
+```
+
+
+The above CRD defines a custom resource called “appdelivery-example-app” of the kind “ExampleApp”. The desired state is captured under the spec section. This example app enables exampleFeature1, disables exampleFeature2, with backup enabled to s3 bucket, “example-backup”. 
+
+The status section is the output given to the operator (i.e. user of the custom resource). This example CRD returns the current version, as well as URL and the last backup time. If you are familiar with the NOTES.txt section of a Helm chart, the status section often contains similar information (e.g. how to connect to the application). 
+
+
+## How KubeSphere uses Kubernetes Operators
+
+The Operator pattern is widely used underneath the hood at KubeSphere. In fact, the core [Logging System](https://kubesphere.io/docs/pluggable-components/logging/) used for log collection, querying, and management is implemented via the [FluentBit Operator](https://github.com/fluent/fluentbit-operator). The log collection is implemented through Fluent Bit, however, vanilla deployment of Fluent Bit can be difficult. To address this challenge, the KubeSphere observability team developed an open-source Fluent Bit management tool called [FluentBit Operator](https://github.com/fluent/fluentbit-operator), which has been donated to the upstream Fluent community. The FluentBit Operator creates and manages Fluent Bit pods through CRDs, and dynamically updates the Fluent Bit configurations and pods, which makes it easier to deploy, configure, and update Fluent Bit instances.
+
+
+## How to Use an Operator in KubeSphere
+
+Using other Operators in KubeSphere is also simple. Log onto the KubeSphere console as admin, and use kubectl from the toolbox in the lower-right hand corner (alternatively, have kube context pointed to the KubeSphere cluster if you have access). 
+
+Next, locate the YAML file that holds all the Operator manifests. For example, for Prometheus Operator, it’s in a file called `[bundle.yaml](https://github.com/prometheus-operator/prometheus-operator/blob/main/bundle.yaml)`:
```

Comment:
`[bundle.yaml](https://github.com/prometheus-operator/prometheus-operator/blob/main/bundle.yaml)`
->
[`bundle.yaml`](https://github.com/prometheus-operator/prometheus-operator/blob/main/bundle.yaml)

To avoid this:
![image](https://user-images.githubusercontent.com/75110798/155941814-4e5dbbf2-6bbe-4b0c-91ce-32c49caa5e60.png)

Reply:
Fixed